views/js/runner/plugins/security/fullScreen.js

```diff
@@ -215,6 +221,12 @@ define([
          * Initializes the plugin (called during runner's init)
          */
         init: function init() {
+            // Since iOS platform restrict keyboard usage in full sreen mode,
+            // do nothing if plugin run under iOS platform
+            if (iOS) {
+                return;
+            }
```

Comment:
Caution! This should not be added before the var declarations, as this will conflict with our coding rules: `always var on top`.

Then, it would be preferable to add a function:
```javascript
function isAllowed() {
    return !iOS;
}
```

And then rely to know if the plugin can be installed. For instance, it could be something like:
```javascript
function init() {
    var testRunner = this.getTestRunner();
    var waitingForUser = false;

    function isAllowed() {
        // Since iOS platform restrict keyboard usage in full sreen mode,
        // do nothing if plugin run under iOS platform
        return !iOS;
    }

    function enableItem() { ... }

    function disableItem() { ... }

    function alertUser() { ... }

    function doPause() { ... }

    if (isAllowed()) {
        // when the runner has just started and the full screen prompt is still displayed, disable the item
        testRunner.after('renderitem.fullscreen', function() { ... });

        testRunner.on('destroy', function() { ... });

        // checks for frame embedding
        if (isFrameEmbedded()) { ... }

        // listen either to the native or the change event created in the observer above
        doc.addEventListener(fullScreenEventName, function() { ... });

        isFullScreen = checkFullScreen();
        if (!isFullScreen) {
            leaveFullScreen(testRunner);
            alertUser();
        } else if (!fullScreenSupported) {
            startFullScreenChangeObserver();
        }
    } else {
        this.disable();
    }
}
```

Reply:
refactored